management-gateway/src/main/java/org/frankframework/management/gateway/SerializableInputStream.java

```diff
@@ -0,0 +1,182 @@
+/*
+   Copyright 2024 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package org.frankframework.management.gateway;
+
+import java.io.Externalizable;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * InputStream wrapper to allow streams to be serialized and deserialized so the (raw) binary data can be sent using Hazelcast.
+ * When deserializing the data is temporarily written to disk. The temporary file is removed once the stream is closed.
+ * 
+ * @author Niels Meijer
+ */
+public class SerializableInputStream extends InputStream implements Externalizable {
+
+	private static final long serialVersionUID = 1L;
+	private static final transient int BUFFERSIZE = 20_480;
+	private static final transient Logger LOG = LogManager.getLogger(SerializableInputStream.class);
+
+	private final transient Path tmpFile;
+
+	private transient InputStream delegate;
+
+	/**
+	 * Constructor for deserialization.
+	 */
+	public SerializableInputStream() {
+		super();
+
+		LOG.trace("creating new SerializableInputStream from deserialization");
+		String tmpdir = System.getProperty("java.io.tmpdir");
+		Path dir = Path.of(tmpdir);
+
+		Path tmpDir = dir.resolve("frank-management-gateway");
```

Comment:
Kan je hier misschien gebruik maken van `Files.createTempDirectory()` en de verschillende parameters?

Reply:
Dat kan vast, in plaats van de `File.createTempFile` neem ik aan?

Reply:
Ja er zijn verschillende combinaties mogelijk. maar wellicht kan je die methoden in `Files` (createTempDirectory/createTempFile) gebruiken, dan hoef je dit niet zelf te doen.